### PR TITLE
v0.0.3 - adds register_bases

### PR DESCRIPTION
**Pull request by spapa013**
**Change Log:**

- adds `register_bases` to enable recursive mapping of base classes to classes in the schema modules using a mirrored directory structure 
   - base classes are applied to both virtual modules (through api/config) and base package modules (through __init__)
- `get_package_version` --> `check_package_version`
- adds `check_latest_version_from_github` to api __init__ to ensure that users are notified when a new version is available.

### CODE REVIEW

File: python/microns-nda-api/microns_nda_api/__init__.py [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
     check_if_latest_kwargs=dict(
         owner='cajal', 
         repo='microns-nda', 
         source='tag', 
     )
-)
\ No newline at end of file
+)
+
+def check_latest_version_from_github(owner='cajal', repo='microns-nda', source='tag', branch=None, path_to_version_file=None, warn=True):
```
Comment (Cpapa97) [inline python/microns-nda-api/microns_nda_api/__init__.py:13]: If this type of wrapper is going to be common you should probably provide a function generator in microns-utils.

Just something simple along the lines of:

```python
# in microns utils / version_utils.py
def generate_latest_version_checker(owner, repo):
    def inner(source='tag', branch=None, path_to_version_file=None, warn=True):
        return check_latest_version_from_github(owner=owner, repo=repo, source=source, branch=branch, path_to_version_file=path_to_version_file, warn=warn) 
    return inner

# then use it here in __init__.py
check_latest_version_from_github = version_utils.generate_latest_version_checker(owner='cajal', repo='microns-nda')
```

File: python/microns-nda/microns_nda/__init__.py [hunk 2]
```diff
@@ -9,3 +9,9 @@
         source='tag', 
     )
 )
+
+def check_latest_version_from_github(owner='cajal', repo='microns-nda', source='tag', branch=None, path_to_version_file=None, warn=True):
```
Comment (Cpapa97) [inline python/microns-nda/microns_nda/__init__.py:13]: Same here.

**Review by Cpapa97 — CHANGES_REQUESTED** [review]

File: python/microns-nda-api/microns_nda_api/__init__.py (hunk 1 above)
Comment (spapa013) [reply python/microns-nda-api/microns_nda_api/__init__.py]: done.

File: python/microns-nda/microns_nda/__init__.py (hunk 2 above)
Comment (spapa013) [reply python/microns-nda/microns_nda/__init__.py]: done

**Pull request merged**